core/src/main/java/com/yahoo/oak/Slice.java

```diff
@@ -251,7 +247,7 @@ public int getLength() {
 
     @Override
     public long getAddress() {
-        return ((DirectBuffer) buffer).address() + getOffset();
+        return memAddress + getOffset();
```

Comment:
So here you return the exact address and not the start of the entire block as written elsewhere in the code. Please check Slice's `getAddress() `usage and update the comments everywhere.